### ISSUE TITLE
Inline error remains after error is fixed

### ISSUE THREAD

**Issue by r0ckwav3** [zed-industries/feedback#602]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

In C++, when Zed finds an error, clicking the error will put it into an inline format. If you then fix the bug, the inline error remains and cannot be removed (as far I could manage).
<img width="230" alt="Screen Shot 2022-10-17 at 12 10 42 PM" src="https://user-images.githubusercontent.com/33732392/196262580-4d1fb8a0-2bc8-4470-a5fa-828171335070.png">
<img width="393" alt="Screen Shot 2022-10-17 at 12 10 52 PM" src="https://user-images.githubusercontent.com/33732392/196262584-ec5be483-920c-43ce-a3de-3e139a330774.png">
<img width="370" alt="Screen Shot 2022-10-17 at 12 11 01 PM" src="https://user-images.githubusercontent.com/33732392/196262589-159ab748-ec2c-42c9-890e-2524bdf45cbf.png">
<img width="352" alt="Screen Shot 2022-10-17 at 12 11 09 PM" src="https://user-images.githubusercontent.com/33732392/196262585-595c3c34-aadc-4857-91c6-53fcdcba9096.png">

(there's another inline error in the second image behind the pop-up, which is my prior attempt at recreating the bug)

### Expected behavior

Either fixing the error would cause the inline error to dissapear or there would be a clear close button to dismiss it.

### Environment

Zed 0.60.4 – /Applications/Zed.app
macOS 12.6
architecture arm64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue

[Zed.log](https://github.com/zed-industries/feedback/files/9803802/Zed.log)

**Comment by mikayla-maki** [comment]
Use the escape key to remove it! 

And yeah, this is the second time we've gotten this bug report, we should make this more discoverable somehow. cc: @iamnbutler

**Comment by JosephTLyons** [comment]
I've been curious to know if more people use the inline errors or the errors in the modal windows.  We currently have 3 different ways to view errors - the aforementioned ways and then project diagnostics.  Is there utility in having both a modal error and inline error?

**Comment by TerminalFi** [comment]
FWIW, I think this could live behind a setting key. I personally would love for these to disappear once resolved and not require a long stretch to esc.

**Comment by iamnbutler** [comment]
> Use the escape key to remove it!
> 
> And yeah, this is the second time we've gotten this bug report, we should make this more discoverable somehow. cc: @iamnbutler

Yeah IMO as others are saying this experience seems broken. It should go away when you have resolved the error. It _feels_ broken. Is there an argument for keeping it the way it is?

**Comment by mikayla-maki** [comment]
> Is there an argument for keeping it the way it is?

It does cause a bit of visual jumping around which might feel unpleasant, especially with regards to scroll position or some such. That said, I don't think the current experience is much better. Probably good to test out how it feels in person, I do like the inline errors....

**Comment by TerminalFi** [comment]
I love the inline errors ! I just think having it user configurable would be best. 

`persist` `delay` `instant`

**Comment by izuzak** [comment]
I regularly ran into this and finally decided to see if it was reported. Yeah, I would never have guessed that `Esc` would remove it. And +1 to removing diagnostic messages which no longer apply i.e. were fixed.

**Comment by izuzak** [comment]
@JosephTLyons 👋 I think I've seen the `diagnostics` label applied to other diagnostics-related issues. Should we apply that here too? 💭

**Comment by JosephTLyons** [comment]
> @JosephTLyons 👋 I think I've seen the `diagnostics` label applied to other diagnostics-related issues. Should we apply that here too? 💭 

Thanks @izuzak, I added the label.  Also, yeah, it is an issue that this is not discoverable.  We need to figure out how to surface a way to dismiss these, and possibly auto-remove them when they are no longer valid.
